src/core/constraints/ShapeBasedConstraint.hpp

```diff
@@ -57,13 +57,13 @@ class ShapeBasedConstraint : public Constraint {
 
 private:
   /** Private methods */
-  void reflect_particle(Particle *p, const double *distance_vector,
+  void reflect_particle_from_flat_wall(Particle *p, const double *distance_vector,
                         const double *folded_pos) const;
 
   /** Private data members */
   std::shared_ptr<Shapes::Shape> m_shape;
 
-  ReflectionType m_reflection_type;
+  int m_reflection_type;
```

Comment:
Please leave the type as it is, and convert the parameters at the python/si level from a string parameter to the corresponding enum val.